### ISSUE TITLE
Toast does not properly support RTL

### ISSUE THREAD

**Issue by alanbsmith** [Workday/canvas-kit#1225]
## 🐛 Bug Report

Toast's System Icon does not have proper bidirectional support. 

It uses [Emotion's `styled` function](https://github.com/Workday/canvas-kit/blob/master/modules/react/toast/lib/Toast.tsx#L4) instead of our `styled` function in `@workday/canvas-kit-react/common`.

**This is something we'll need to fix in 4.x and 5.x**

## To Reproduce

[View this sandbox](https://codesandbox.io/s/ck-5-1-0-toast-rtl-devzi?file=/src/App.tsx)

## Expected Behavior

`marginRight` should properly respond to the `ContentDirection` provided in the theme.

## Actual Results

The `marginRight` is fixed.

## Browser (if applicable)

n/a

## Link to repl or repo (highly encouraged)

[Sandbox](https://codesandbox.io/s/ck-5-1-0-toast-rtl-devzi?file=/src/App.tsx)

### Error Output

![Screen Shot 2021-08-16 at 23 14 10](https://user-images.githubusercontent.com/4818182/129667415-c3da5343-1531-44db-91f6-1e5c36c6961c.png)

## Acceptance Criteria

- [ ] Fix the `marginRight` style bug so that it has proper bidirectional support.
- [ ] Add a [RTL Toast story example](https://github.com/Workday/canvas-kit/blob/master/modules/react/toast/stories/toast.stories.mdx) to our docs
- [ ] Add a [visual regression test](https://github.com/Workday/canvas-kit/blob/master/modules/react/toast/stories/stories_visualTesting.tsx)
